controllers/volumereplicationgroup_controller.go

```diff
@@ -258,6 +264,31 @@ func filterPVC(mgr manager.Manager, pvc *corev1.PersistentVolumeClaim, log logr.
 	return req
 }
 
+func GetPVCLabelSelector(
```

Comment:
This function return a label selector that corresponds to the recipe if one is used. However, if no recipe is used, it will return the regular pvcSelector. Therefore, if a mix of resources belonging to a workload use recipes and some don't, the resulting label selector from this function call may not cover all PVCs.
I guess to question comes to; can you have an application that is a mix of resources, where some resources uses a recipe and some other don't use a recipe?

Reply:
Right now, we don't have a case where combining pvcSelectors and volume groups is a known requirement. That's not to say we're not interested in it; but combining the labelSelectors is a bit complex. Specifically, there are some labels that create "include" behavior and others that create "exclude" behavior, and combining them without specifically knowing the intent of the user will cause problems. We had a design for this at some point, where we'd add specific fields to the API that would signify the intent, but that was dropped in favor of the simpler case. Right now, you just pick one: pvcSelector or the volume group label selector, and the volume group label selector has precedence. If we find a case where we need to combine them, I can create an issue. For now, the unit tests reflect the "one or the other" design.